panel.py

```diff
@@ -0,0 +1,376 @@
+"""
+Module representing detector.
+Creates a detector list from a geometry file (crystfel type) and
+matrix size for the image.
+"""
+
+import numpy as np
+
+from stream_read import dict_peak_in_name_h5
+
+
+class Detector:
+    """
+    Class representing a detector.
+    min_fs - min index in a column
+    min_ss - min index in a row
+    max_fs - max index in a column
+    max_ss - max index in a row
+    corner_x - coordinates of the panel corner from geom file
+    corner_y - coordinates of the panel corner from geom file
+    array - data from each panel from part of the data - dataset in h5 file
+    position - panel coordinates on the final image
+    peak_list - list of peaks from the stream file
+    peak_near_bragg_list - peak list from 'this' (?) panel like the
+        check-near-bragg script does.
+    """
+    def __init__(self, image_size, name, min_fs, min_ss, max_fs, max_ss, xfs,
+                 yfs, xss, yss, corner_x, corner_y, data):
+        self.name = name
+        self.min_fs = min_fs
+        self.min_ss = min_ss
+        self.max_fs = max_fs
+        self.max_ss = max_ss
+        self.xfs = xfs
+        self.yfs = yfs
+        self.xss = xss
+        self.yss = yss
+        self.corner_x = corner_x
+        self.corner_y = corner_y
+        self.array = np.copy(data[self.min_ss: self.max_ss + 1,
+                             self.min_fs: self.max_fs + 1])
+        # my position in matrix
+        self.position = (0, 0)
+        self.peak_list = []
+        self.peak_near_bragg_list = []
+        self.image_size = image_size
+
+    def get_peak_list(self):
+        return self.peak_list
+
+    def get_peak_near_bragg_list(self):
+        return self.peak_near_bragg_list
+
+    def get_array_rotated(self):
+        """
+        Returns array size for each panel after rotation.
+        """
+        self.with_rotatioon()
+        return self.array
+
+    def with_rotatioon(self):
+        """
+        By comparing xfs, yfs, xss and yss verifies which kind of rotation
+        should be applied.
+        """
+        if np.abs(self.xfs) < np.abs(self.yfs) and \
+                np.abs(self.xss) > np.abs(self.yss):
+            if self.yfs > 0 and self.xss < 0:
+                self.rot_y_x()
+            elif self.yfs < 0 and self.xss > 0:
+                self.rot_y_2x()
+        elif np.abs(self.xfs) > np.abs(self.yfs) and \
+                np.abs(self.xss) < np.abs(self.yss):
+            if self.xfs < 0 and self.yss < 0:
+                self.rot_y()
+            elif self.xfs > 0 and self.yss > 0:
+                self.rot_x()
+        else:
+            print(self.name, "Unknown rotation!")
+            exit()
+
+    def rot_x(self):
+        """
+        Rotation along x axis, columns stay the same, rows are switched.
+        """
+        for peak_stream in self.peak_list:
+            # to sa peaki które wyświetla check peak detection
```

Comment:
all comments in this file are in polish, this has to be translated

Reply:
Addressed in: 9a3ec8fbb4bf6a7e079bb717985c048545ccc3e6